crates/ruff/src/rules/tryceratops/rules/raise_within_try.rs

```diff
@@ -83,9 +87,54 @@ pub(crate) fn raise_within_try(checker: &mut Checker, body: &[Stmt], handlers: &
         visitor.raises
     };
 
+    if raises.is_empty() {
+        return;
+    }
+
+    // The names of exceptions handled by the `try` Stmt. We can't compare `Expr`'s since they have
+    // different ranges, but by virtue of this function's call path we know that the `raise`
+    // statements will always be within the surrounding `try`.
+    let handler_ids: Vec<&str> = helpers::extract_handled_exceptions(handlers)
+        .into_iter()
+        .filter_map(|handler| {
+            if let Expr::Name(ast::ExprName { id, .. }) = handler {
+                Some(id.as_str())
+            } else {
+                None
+            }
+        })
+        .collect();
+
     for stmt in raises {
-        checker
-            .diagnostics
-            .push(Diagnostic::new(RaiseWithinTry, stmt.range()));
+        let Stmt::Raise(ast::StmtRaise { exc: Some(exception), .. }) = stmt else {
+            continue;
+        };
+
+        let Some(exc_name) = get_function_name(exception.as_ref()) else {
+            continue;
+        };
+
+        // We can't check exception sub-classes without a type-checker implementation, so let's
+        // just catch the blanket `Exception` for now.
+        if (handler_ids.contains(&"Exception") && checker.semantic().is_builtin("Exception"))
+            || handler_ids.contains(&exc_name)
```

Comment:
Alternatively:

```rust
if handler_exprs.contains(ComparableExpr::from(map_callable(exception)) || handler_exprs.iter().any(|expr| checker.resolve().resolve_call_path(expr).map_or(false, |call_path| matches!(call_path.as_slice(), ["", "Exception" | "BaseException"]) {
  ...
}
```

`map_callable` will take the `func` if the expression is a call, or return the expression otherwise (which will also ensure we handle `raise MyException` (no parens required technically)). Using `ComparableExpr` ensures we handle attributes (`raise foo.Bar()`).